main/adapters/handlers/signer.go

```diff
@@ -272,9 +272,17 @@ func getHTTPResponse(statusCode int, signingResp *signingResponse) h.HTTPRespons
 		log.Warnf("error serializing signing response: %v", err)
 	}
 
+	headers := http.Header{"Content-Type": {h.JSONType}}
+	if statusCode != http.StatusOK {
```

Comment:
It is possible for the `Response` field of the `signingResp` to be a nil pointer which would cause a runtime error if we try to access `signingResp.Response.Header`, so to be safe, we can check this here

```suggestion
	if signingResp.Response != nil && statusCode != http.StatusOK {
```